flexmeasures/auth/policy.py

```diff
@@ -197,3 +197,50 @@ def check_account_role(user, principal: str) -> bool:
         if user.account.has_role(account_role):
             return True
     return False
+
+
+def can_modify_role(user, roles_to_modify) -> bool:
+    """For a set of supported roles, check if the current user can modify the roles.
+
+    :param user: The current user.
+    :param roles_to_modify: A list of roles to modify.
+    :return: True if the user can modify the roles, False otherwise.
+
+    The roles are:
+    - admin: can only be changed in CLI / directly in the DB
+    - admin-reader: can be added and removed by admins
+    - account-admin: can be added and removed by admins and consultants
+    - consultant: can be added and removed by admins and account-admins
+
+    """
+
+    statuses = []
+
+    for role in roles_to_modify:
+
+        if isinstance(role, int):
+            from flexmeasures.data.models.user import Role
+
+            role = current_app.db.session.get(Role, role)
+
+        if not role:
+            statuses.append(False)
+            continue
+        if role.name == ADMIN_ROLE:
+            statuses.append(False)
+            continue
+        if role.name == ADMIN_READER_ROLE and not user.has_role(ADMIN_ROLE):
+            statuses.append(False)
+            continue
+        if role.name == ACCOUNT_ADMIN_ROLE and not (
+            user.has_role(ADMIN_ROLE) or user.has_role(CONSULTANT_ROLE)
+        ):
+            statuses.append(False)
+            continue
+        if role.name == CONSULTANT_ROLE and not (
+            user.has_role(ADMIN_ROLE) or user.has_role(ACCOUNT_ADMIN_ROLE)
+        ):
+            statuses.append(False)
+            continue
+
+    return all(statuses)
```

Comment:
If you don't have a good reason, I would suggest we don't need `statuses` and we can simply directly `return False` whenever a check fails. And `return True` in line 246.